### PR TITLE
[NETBEANS-5599] PHP 8.1 Support

### PR DESCRIPTION
**Pull request by junichi11**
https://issues.apache.org/jira/browse/NETBEANS-5599

- https://wiki.php.net/rfc/final_class_const
- https://wiki.php.net/rfc/noreturn_type
- https://wiki.php.net/rfc/readonly_properties_v2
- https://wiki.php.net/rfc/new_in_initializers
- https://wiki.php.net/rfc/pure-intersection-types

## Note:

- This implementation has not finished yet. (e.g. enum, and so on)
- Parts of features are not fixed yet. (e.g. Hints, CC)

### CODE REVIEW

**Comment by junichi11** [comment]
Will merge this after #3461 is merged.

**Review by tmysik — APPROVED** [review]

**Comment by tmysik** [comment]
@junichi11 Approved, of course. Please, let me know if you are going to merge it or whether I should do it. Thank you!

**Comment by junichi11** [comment]
@tmysik Please merge it. Thank you!

**Pull request merged**
